### PR TITLE
Documenting the select and subquery strategy

### PR DESCRIPTION
**Pull request by lorenzo**
improvement over https://github.com/cakephp/docs/pull/2711 and https://github.com/cakephp/docs/pull/2710

### CODE REVIEW

File: en/orm/retrieving-data-and-resultsets.rst [hunk 1]
```diff
@@ -589,6 +589,72 @@ association properties in your results.
...
+    $articles->FirstComment->strategy('select');
+
+Using the ``select`` strategy is also a great way of making associations with
+tables in another database, since it would not be possible to fetch records
+using ``joins``.
+
+Fetching With The Subquery Strategy
+-----------------------------------
+
+As your tables grow in since, fetching associations from them can become
```
Comment (dereuromark) [inline en/orm/retrieving-data-and-resultsets.rst]: size

File: en/orm/retrieving-data-and-resultsets.rst [hunk 2]
```diff
@@ -589,6 +589,72 @@ association properties in your results.
...
+                    return $q->where(['Comments.approved' => true]);
+                }
+        ]
+    ]);
+
+The result will remain the same as with using the default strategy, but this
+can greatly improve the query and fetching time in some databases, in
+particular it will allow to fetch big chunks of data at the same time in
+databases that limit the amount of bound parameters per query, such as
+**Microfost SQL Server**.
```
Comment (dereuromark) [inline en/orm/retrieving-data-and-resultsets.rst]: fost vs soft :)

**Comment by markstory** [comment]
Typos fixed up.

File: en/orm/retrieving-data-and-resultsets.rst [hunk 3]
```diff
@@ -589,6 +589,72 @@ association properties in your results.
...
+                    return $q->where(['Comments.approved' => true]);
+                }
+        ]
+    ]);
+
+The result will remain the same as with using the default strategy, but this
+can greatly improve the query and fetching time in some databases, in
+particular it will allow to fetch big chunks of data at the same time in
+databases that limit the amount of bound parameters per query, such as
+**Microsost SQL Server**.
```
Comment (dereuromark) [inline en/orm/retrieving-data-and-resultsets.rst]: Do you on purpose misspell Microsoft? :D

Comment (markstory) [inline en/orm/retrieving-data-and-resultsets.rst]: No, I'm just bad at typing/reading.

Comment (markstory) [inline en/orm/retrieving-data-and-resultsets.rst]: Actually fixed now.

**Comment by antograssiot** [comment]
looks good, you can merge it if you want, I'll do the translation later in the week if no one does.

**Pull request merged**